src/heritedcode_api_functions.py

```diff
@@ -0,0 +1,98 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# heritedcode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/heritedcode for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+import requests
+import json
+
+
+class ApiResponse:
+    """
+    this class represents response data structure for returning response
+    from functions.
+    """
+
+    def __init__(
+        self,
+        checksums,
+        data_url,
+        filetype_url,
+        language_url,
+        length,
+        license_url,
+        status,
+    ):
+        """represents diffrent type of checksums like sha1,sha1_git and their values"""
+        self.checksums = checksums
+        """represents data url for available raw data of any code"""
+        self.data_url = data_url
+        """represents url to get mime type of a file like text/x-c etc."""
+        self.filetype_url = filetype_url
+        """represents url to get language of a file like text/c/c++ etc."""
+        self.language_url = language_url
+        """represents length of the file"""
+        self.length = length
+        """represents url to get license of a file"""
+        self.license_url = license_url
+        self.status = status
+
+    def __eq__(self, second_class_obj):
+        if isinstance(second_class_obj, ApiResponse):
+            for key in vars(self):
+                if getattr(self, key) != getattr(second_class_obj, key):
+                    return False
+            return True
+        return False
+
+    """
+    Returns an instance of ApiResponse class.
+    """
+
+    @classmethod
+    def from_swh(
+        cls,
+        checksums,
+        data_url,
+        filetype_url,
+        language_url,
+        length,
+        license_url,
+        status,
+    ):
+        return cls(
+            checksums, data_url, filetype_url, language_url, length, license_url, status
```

Comment:
Always prefer using named keyword arguments over positional arguments: this is more robust and avoid problems if arguments are renamed and moved in the future:

```suggestion
            checksums=checksums,
            data_url=data_url,
            filetype_url=filetype_url,
            language_url=language_url,
            length=length,
            license_url=license_url,
            status=status,
```